dape.el

```diff
@@ -2151,13 +2151,16 @@ symbol `dape-connection'."
                      (cl-map 'list 'identity
                              (plist-get config 'command-args)))))
           (setq server-process
-                (make-process :name "dape adapter"
-                              :command command
-                              :filter (lambda (_process string)
-                                        (dape--repl-message string))
-                              :noquery t
-                              :file-handler t
-                              :stderr stderr-buffer))
+                (let ((process-environment (cl-copy-list process-environment)))
+                  (cl-loop for (key value) on (plist-get config 'command-env) by 'cddr
+                           do (setenv key (format "%s" value)))
```

Comment:
I think it makes sense to let type of `key` be valid for string, symbols and keyword.
```emacs-lisp
(setenv (pcase key
          ((pred keywordp) (substring (format "%s" key) 1))
          ((or (pred symbolp) (pred stringp)) (format "%s" key))
          (_ (user-error "Bad type for `command-env' key %S" key)))
        (format "%s" key))
```